raft.c

```diff
@@ -510,17 +517,42 @@ static int raftApplyLog(raft_server_t *raft, void *user_data, raft_entry_t *entr
 
                 int e = raft_recv_entry(rr->raft, rem_entry, &resp);
                 assert (e == 0);
-
                 raft_entry_release(rem_entry);
+
                 break;
             }
         case RAFT_LOGTYPE_REMOVE_NODE:
             req = (RaftCfgChange *) entry->data;
+
+            entry->user_data = NULL;
+            rr->client_attached_entries--;
+
+            if (raft_req) {
+                RedisModule_ReplyWithSimpleString(raft_req->ctx, "OK");
+                RaftReqFree(raft_req);
+            }
+
             if (req->id == raft_get_nodeid(raft)) {
                 LOG_DEBUG("Removing this node from the cluster");
+                rr->raft_exit_node = true;
                 return RAFT_ERR_SHUTDOWN;
             }
             break;
+        case RAFT_LOGTYPE_ADD_NONVOTING_NODE:
+            entry->user_data = NULL;
+            rr->client_attached_entries--;
```

Comment:
Looks like this is repeating, maybe an `entryDetachRaftReq()` would be useful?

Reply:
done